src/EditorFeatures/CSharpTest/CodeActions/ExtractMethod/ExtractMethodTests.cs

```diff
@@ -498,7 +498,7 @@ class Program
     static void Main()
     {
         byte z = 0;
-        Goo<byte, byte>({|Rename:NewMethod|}(), y => { return 0; }, z, z);
+        Goo({|Rename:NewMethod|}(), y => { return (byte)0; }, z, z);
```

Comment:
@DustinCampbell see https://github.com/dotnet/roslyn/pull/26237/files#r182592374

Now, we attempt to simplify the name first rather than the cast first, so we get this behavior.  IMO, i think the new code is preferable to the old.  Using inference for generic methods is very preferred (especially as you might end up having to name a lot of type parameters), so i think we should attempt to preserve inference first, even if it means we have to keep some casts.